packages/tailwindcss/src/utilities.ts

```diff
@@ -2785,6 +2792,699 @@ export function createUtilities(theme: Theme) {
     position: (value) => [gradientStopProperties(), decl('--tw-gradient-to-position', value)],
   })
 
+  if (enableMaskUtilities) {
+    /**
+     * @css `mask-image`
+     */
+
+    staticUtility('mask-none', [['mask-image', 'none']])
+
+    utilities.functional('mask', (candidate) => {
+      if (!candidate.value) return
+      if (candidate.modifier) return
+      if (candidate.value.kind !== 'arbitrary') return
+
+      // Arbitrary values
+      let value: string | null = candidate.value.value
+      let type =
+        candidate.value.dataType ??
+        inferDataType(value, ['image', 'percentage', 'position', 'bg-size', 'length', 'url'])
+
+      switch (type) {
+        case 'percentage':
+        case 'position': {
+          if (candidate.modifier) return
+          return [decl('mask-position', value)]
+        }
+        case 'bg-size':
+        case 'length':
+        case 'size': {
+          return [decl('mask-size', value)]
+        }
+        case 'image':
+        case 'url':
+        default: {
```

Comment:
We don't have tests for the `image` and `url` data types

Reply:
We do:

<img width="400" alt="Screenshot 2025-03-28 at 07 15 42" src="https://github.com/user-attachments/assets/44385d5d-7019-453d-94c8-36762399d28f" />

Reply:
Ah wait I bet you mean when inferred not explicitly given

Reply:
Yep exactly